### PR TITLE
feat: Add hasura infra for discord bot

### PR DESCRIPTION
**Pull request by karelianpie**
## Description

* Add event trigger `discordUserLinked`;

## Screenshots (if appropriate):

```cmd
==========================================================
Development server has started successfully!
Visit http://localhost:3000 to view the Coordinape app.
==========================================================
{
  fn: 'handleDiscordUserAdded',
  payload: {
    created_at: '2022-11-22T16:05:39.94684Z',
    delivery_info: { current_retry: 0, max_retries: 5 },
    event: {
      data: [Object],
      op: 'INSERT',
      session_variables: [Object],
      trace_context: [Object]
    },
    id: '8b7b4f1a-e42d-4e7b-80f5-609a9bddd750',
    table: { name: 'users', schema: 'discord' },
    trigger: { name: 'discordUserLinked' }
  }
}
```

### CODE REVIEW

File: hasura/metadata/actions.graphql [hunk 1]
```diff
@@ -222,7 +226,6 @@ input UpdateCircleInput {
   update_webhook: Boolean
   fixed_payment_token_type: String
   fixed_payment_vault_id: Int
-  show_pending_gives: Boolean
```
Comment (topocount) [inline hasura/metadata/actions.graphql:225]: why get rid of this in this PR?

File: hasura/metadata/databases/default/tables/public_vaults.yaml [hunk 2]
```diff
@@ -23,18 +23,6 @@ array_relationships:
...
-      hasura_fields:
-        - token_address
-        - chain_id
-      remote_field:
-        price_per_share:
-          arguments:
-            chain_id: $chain_id
-            token_address: $token_address
-      remote_schema: vaults
-    name: price_per_share
```
Comment (topocount) [inline hasura/metadata/databases/default/tables/public_vaults.yaml:37]: this also probably shouldn't be deleted. perhaps this needs to be rebased?

File: hasura/metadata/remote_schemas.yaml [hunk 3]
```diff
@@ -1,5 +1 @@
-- name: vaults
-  definition:
-    url: '{{HASURA_API_BASE_URL}}/remote/vaults'
-    timeout_seconds: 60
-  comment: ""
```
Comment (topocount) [inline hasura/metadata/remote_schemas.yaml:5]: same as above

File: hasura/metadata/databases/default/tables/public_vaults.yaml (hunk 2 above)
Comment (karelianpie) [reply hasura/metadata/databases/default/tables/public_vaults.yaml:37]: We can talk about this later today, I just ran `yarn generate` and this was the result

Comment (levity) [reply hasura/metadata/databases/default/tables/public_vaults.yaml:37]: You might have just needed to re-run `yarn hasura metadata reload` again with the dev server running before running generate

Comment (karelianpie) [reply hasura/metadata/databases/default/tables/public_vaults.yaml:37]: > You might have just needed to re-run `yarn hasura metadata reload` again with the dev server running before running generate

Tried that now with the same result, topo mentioned yesterday that he'll have a look at the generate script 🙇

File: api-lib/event_triggers/discordUserLinked.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  try {
+    const payload: EventTriggerPayload<'discord_users', 'INSERT'> = req.body;
+
+    const linked = await handleDiscordUserAdded(payload);
+    res
+      .status(200)
+      .json({ message: `Discord user ${linked ? 'linked' : 'not linked'}` });
+  } catch (e) {
+    res.status(500).json({
+      error: '500',
```
[the remark this answers is not in the document]
Comment (karelianpie) [reply api-lib/event_triggers/discordUserLinked.ts]: The codebase still has a few of these, refactored with https://github.com/coordinape/coordinape/pull/1687/commits/a3dfd6eb6316d766b95a3e07bc334af78ce5d546

File: hasura/metadata/databases/default/tables/public_vaults.yaml (hunk 2 above)
Comment (karelianpie) [reply hasura/metadata/databases/default/tables/public_vaults.yaml]: Ended up fixing these manually with https://github.com/coordinape/coordinape/pull/1687/commits/9231d3344b1ebd0eda28693a8f9ab06b9673b416 and https://github.com/coordinape/coordinape/pull/1687/commits/290da405d9db9eec684d05f7c59634c51d5de769 as the generate script is still not working for me

File: hasura/metadata/actions.graphql (hunk 1 above)
Comment (karelianpie) [reply hasura/metadata/actions.graphql]: These were generated by the generate script as discussed, fixed manually for now with https://github.com/coordinape/coordinape/pull/1687/commits/9231d3344b1ebd0eda28693a8f9ab06b9673b416 and https://github.com/coordinape/coordinape/pull/1687/commits/290da405d9db9eec684d05f7c59634c51d5de769

File: hasura/metadata/remote_schemas.yaml (hunk 3 above)
Comment (karelianpie) [reply hasura/metadata/remote_schemas.yaml]: https://github.com/coordinape/coordinape/pull/1687/commits/9231d3344b1ebd0eda28693a8f9ab06b9673b416 and https://github.com/coordinape/coordinape/pull/1687/commits/290da405d9db9eec684d05f7c59634c51d5de769

**Review by topocount — APPROVED** [review]
This looks good on it's face. I'm not sure how the event trigger will be used now that we're using a GQL subscription in the bot context, but the cost of deleting that scaffolding before launch is minimal.

**Pull request merged**